e2e/host/zonefile

```diff
@@ -10,24 +10,29 @@
 
 $TTL 1h
 
-example.com.                IN A 172.20.10.2
+example.com.                                     IN A 172.20.10.2
 
 ; Host records
 
-to.host.host.example.com.         IN A 172.20.10.2
-_redirect.to.host.host.example.com.          IN TXT "v=txtv0;to=https://to-redirect.host.host.example.com;type=host;code=302" 
+to.host.host.example.com.                        IN A 172.20.10.2
+_redirect.to.host.host.example.com.              IN TXT "v=txtv0;to=https://to-redirect.host.host.example.com;type=host;code=302" 
 
-nocode.host.host.example.com.    IN A 172.20.10.2
-_redirect.nocode.host.host.example.com.      IN TXT "v=txtv0;to=https://nocode.host.host.example.com;type=host"
+nocode.host.host.example.com.                    IN A 172.20.10.2
+_redirect.nocode.host.host.example.com.          IN TXT "v=txtv0;to=https://nocode.host.host.example.com;type=host"
 
-noversion.host.host.example.com. IN A 172.20.10.2
-_redirect.noversion.host.host.example.com.   IN TXT "to=https://noversion.host.host.example.com;type=host"
+noversion.host.host.example.com.                 IN A 172.20.10.2
+_redirect.noversion.host.host.example.com.       IN TXT "to=https://noversion.host.host.example.com;type=host"
 
-noto.host.host.example.com.      IN A 172.20.10.2
-_redirect.noto.host.host.example.com.        IN TXT "v=txtv0;type=host"
+noto.host.host.example.com.                      IN A 172.20.10.2
+_redirect.noto.host.host.example.com.            IN TXT "v=txtv0;type=host"
 
-referer.host.host.example.com.   IN A 172.20.10.2
-_redirect.referer.host.host.example.com.     IN TXT "to=https://referer-redirect.host.host.example.com;type=host;ref=true"
+referer.host.host.example.com.                   IN A 172.20.10.2
+_redirect.referer.host.host.example.com.         IN TXT "to=https://referer-redirect.host.host.example.com;type=host;ref=true"
 
-fallback.host.host.example.com.              IN A 172.20.10.2
-_redirect.fallback.host.host.example.com.    IN TXT "v=txtv0;to=https://{label30};type=host;code=302" 
+fallback.host.host.example.com.                  IN A 172.20.10.2
+_redirect.fallback.host.host.example.com.        IN TXT "v=txtv0;to=https://{label30};type=host;code=302" 
+
+sourcerecord.host.host.example.com.              IN A 172.20.10.2
+upstreamrecord.host.host.example.com.            IN A 172.20.10.2
+_redirect.sourcerecord.host.host.example.com.    IN TXT "v=txtv0;to=https://wrongredirect.host.host.example.com;type=host;code=302;use=_redirect.upstreamrecord.host.host.example.com" 
```

Comment:
Interesting edge case that `to=` and `use=` are used at the same time. So `use=` takes precedent. Seems valid.

Reply:
Generally I feel like we should separately test this.